api/services/Domain.js

```diff
@@ -111,6 +112,68 @@ function checkAcmeChallengeDnsRecord(domain) {
   );
 }
 
+/**
+ * @param {SiteModel} site
+ * @param {DomainModel[]} domains
+ * @param {string} context Domain.Contexts.Site or DomainContexts.Demo
+ */
+function isSiteUrlManagedByDomain(site, domains, context) {
+  const siteDomain = Site.domainFromContext(context);
+  if (site[siteDomain] === null) { return true; }
+  if (domains.length === 0) { return false; }
+  const siteUrlIsManagedByDomain = domains
+    .filter(domain => domain.context === context)
+    .some(domain => domain.namesArray()
+      .some(name => `https://${name}` === site[siteDomain]));
+  return siteUrlIsManagedByDomain;
+}
+
+/**
+ * @param {DomainModel} domain
+ * @param {SiteModel} site
+ */
+async function rebuildAssociatedSite(domain, site) {
+  const branch = site[Site.branchFromContext(domain.context)];
+  if (branch) {
+    await Build.create({
+      site: site.id,
+      branch,
+      username: 'Federalist Operators',
+    }).then(b => b.enqueue());
+  }
+}
+
+/**
+ * @param {DomainModel} domain
+ * @returns {Promise<DomainModel>}
+ */
+async function updateSiteForProvisionedDomain(domain) {
+  // Populate appropriate site URL if it isn't already set
+  const site = await domain.getSite();
+  const firstDomain = `https://${domain.firstName()}`;
+  const siteDomain = Site.domainFromContext(domain.context);
+  if (isSiteUrlManagedByDomain(site, [domain], domain.context) && site[siteDomain] === null) {
+    await site.update({ [siteDomain]: firstDomain });
+    await module.exports.rebuildAssociatedSite(domain, site);
+  }
+  return domain;
+}
+
+/**
+ * @param {DomainModel} domain
+ * @returns {Promise<DomainModel>}
+ */
+async function updateSiteForDeprovisionedDomain(domain) {
+  // Clear appropriate site URL if it matches a domain name
+  const site = await domain.getSite();
+  if (isSiteUrlManagedByDomain(site, [domain], domain.context)) {
+    const siteDomain = domain.context === Domain.Contexts.Site ? 'domain' : 'demoDomain';
```

Comment:
```suggestion
    const siteDomain = Site.domainFromContext(domain.context);
```

Reply:
Missed a few spots. Thanks for catching them!